### DISCUSSION TITLE
Test Fail `"./lib/foundry-huff/scripts/binary_check.sh": No such file or directory (os error 2)`

### DISCUSSION THREAD

**Discussion by 0xbadal** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#5] [Q&A]
I am getting this error. current OS is Windows

```solidity
pragma solidity 0.8.20;

import {Test, console} from "lib/forge-std/src/Test.sol";
import {Base_Test, HorseStore} from "./Base_Test.t.sol";
import {HuffDeployer} from "lib/foundry-huff/src/HuffDeployer.sol";

contract HorseStoreV1Solc is Base_Test {
    function setUp() public override {
        // horseStore = // Huff Edition
        super.setUp();
        horseStoreHuff = HorseStore(HuffDeployer.config().deploy(horseStoreLocation));
    }

    function testReadHorseStore() public {
        uint256 intialValue = horseStoreSol.readNumberOfHorses();
        assertEq(intialValue, 0);
    }
}
```
ERROR
 ``` bash 
forge test --match-path *Huff*
[⠒] Compiling...
No files changed, compilation skipped
2024-03-24T01:21:26.589495Z ERROR forge::runner: setUp failed reason=failed to execute command cd "/mnt/c/Users/HP/Documents/Hacking Notes/WEB3/program/Projects/Horse-Store" && "./lib/foundry-huff/scripts/binary_check.sh": No such file or directory (os error 2) contract=0x34A1D3fff3958843C43aD80F30b94c510645C316

Ran 1 test for test/V1/HorseStoreV1Huff.t.sol:HorseStoreV1Solc
[FAIL. Reason: setup failed: failed to execute command cd "/mnt/c/Users/HP/Documents/Hacking Notes/WEB3/program/Projects/Horse-Store" && "./lib/foundry-huff/scripts/binary_check.sh": No such file or directory (os error 2)] setUp() (gas: 0)
Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 11.37ms (0.00ns CPU time)

Ran 1 test suite in 56.51ms (11.37ms CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/V1/HorseStoreV1Huff.t.sol:HorseStoreV1Solc
[FAIL. Reason: setup failed: failed to execute command cd "/mnt/c/Users/HP/Documents/Hacking Notes/WEB3/program/Projects/Horse-Store" && "./lib/foundry-huff/scripts/binary_check.sh": No such file or directory (os error 2)] setUp() (gas: 0)

Encountered a total of 1 failing tests, 0 tests succeeded
```

**Comment by cromewar** [comment]
Hello, @Badal-Sharma-09; thank you for posting this comment.

The problem resides in your computer not being able to find the correct path on which the dependencies live.
I will assume you are using WSL based on the "/mnt/" path on your error code.

You are pointing to a local folder on the Windows directory: "/mnt/c/Users/HP/Documents/Hacking Notes/WEB3/program/Projects/Horse-Store" and a Linux WSL folder path: "./lib/foundry-huff/scripts/binary_check.sh".

I would say that this happens because of the way WSL and Windows File system works and how the connection between them is "sloppy" at best. I would suggest you please clone the repo/project not to a documents folder on your Windows file system but on the Linux one directly instead and try again.

So basically, just open your WSL terminal, which by default will open at "~", create a folder on it if you will, and clone the repo again, then try doing exactly the same and let us know if that actually works.

**Reply by 0xbadal** [reply]
Thank you, brother. it was helpful.

**Comment by swaploard** [comment]
@cromewar No, it does not work for me. I have created a whole new project and installed all the dependencies, yet I am still facing the same error.

**Reply by EngrPips** [reply]
Hello @swaploard, Try to manually provide the path to the test file and see if the error persists.
